src/agent/rustjail/src/mount.rs

```diff
@@ -1224,4 +1266,80 @@ mod tests {
 
         assert!(check_proc_mount(&mount).is_err());
     }
+    #[test]
+    fn test_rootfs_not_exists() {
+        let rootfs = "/home/rootfs";
+        assert_eq!("/home/rootfs/a/b/c", secure_join(rootfs, "a/b/c"));
+    }
+
+    #[test]
+    fn test_rootfs_relative() {
+        let rootfs = "/home/rootfs";
+        assert_eq!("/home/rootfs/a/b/c", secure_join(rootfs, "../../../a/b/c"));
+    }
+
+    #[test]
+    fn test_rootfs_strip_any_relative() {
+        let rootfs = "/home/rootfs";
+        assert_eq!(
+            "/home/rootfs/a/b/c",
+            secure_join(rootfs, "../../../a/../../b/../../c")
+        );
+    }
+
+    #[test]
+    fn test_rootfs_relative_softlink_1() {
+        let rootfs = "/tmp/rootfs1";
+        // /var/run/1 -> ../../../../
+        let unsafe_path = "/var/run/1";
+
+        // prepare softlink to test
+        remove_dir_all("/tmp/rootfs1").unwrap_or_default();
+        create_dir_all("/tmp/rootfs1/var/run").expect("failed to create rootfs");
+        fs::symlink("../../../../", "/tmp/rootfs1/var/run/1").unwrap();
+
+        assert_eq!("/tmp/rootfs1", secure_join(rootfs, unsafe_path));
+    }
+
+    #[test]
+    fn test_rootfs_relative_softlink_2() {
+        let rootfs = "/tmp/rootfs2";
+        // /var/run/2 -> ..
+        let unsafe_path = "/var/run/2";
+
+        // prepare softlink to test
+        remove_dir_all("/tmp/rootfs2").unwrap_or_default();
+        create_dir_all("/tmp/rootfs2/var/run").expect("failed to create rootfs");
+        fs::symlink("..", "/tmp/rootfs2/var/run/2").unwrap();
+
+        assert_eq!("/tmp/rootfs2/var", secure_join(rootfs, unsafe_path));
+    }
+
+    #[test]
+    fn test_rootfs_relative_softlink_3() {
+        let rootfs = "/tmp/rootfs3";
+        // /var/run/3 -> ../../../dd
+        let unsafe_path = "/var/run/3";
+
+        // prepare softlink to test
+        remove_dir_all("/tmp/rootfs3").unwrap_or_default();
+        create_dir_all("/tmp/rootfs3/var/run").expect("failed to create rootfs");
+        fs::symlink("../../../dd", "/tmp/rootfs3/var/run/3").unwrap();
+
+        assert_eq!("/tmp/rootfs3/dd", secure_join(rootfs, unsafe_path));
+    }
+
+    #[test]
+    fn test_rootfs_absolute_softlink() {
+        let rootfs = "/tmp/rootfs4";
+        // /var/run/4 -> /
+        let unsafe_path = "/var/run/4";
+
+        // prepare softlink to test
+        remove_dir_all("/tmp/rootfs4").unwrap_or_default();
+        create_dir_all("/tmp/rootfs4/var/run").expect("failed to create rootfs");
+        fs::symlink("/", "/tmp/rootfs4/var/run/4").unwrap();
```

Comment:
Please can you add appropriate calls to `skip_if_not_root()` / `skip_if_root()` for these tests.

Reply:
Either root or non-root user is Ok